### DISCUSSION TITLE
Problems with loggin in even though password is correct |  generates wrong codes

### DISCUSSION THREAD

**Discussion by ZupflKnoe568del** [2fast-team/2fast#178] [Q&A] [answered]
Hi Jan Philipp, I like the 2fast authenticator a lot: it´s intuitive, powerful and clearly laid out. I have version 1.3.2.0 and Windows 11 pro, version 23H2, latest updates. 

Unfortunately, it sometimes happens that I need **three** attempts to login into it, even though I punch in the password correctly. What´s worse: it generates **wrong** 6-digit codes with two important accounts of mine. I had to fall back on my smartphone authenticator but I prefer your authenticator on my PC. Sometimes an error message pops up with something like "a massive error ocurred, please copy and send the log to GITHUB. 

Actually I am a subscriber to your 2fast authenticator and have tried to send you money via Microsoft, but I doubt it ever reached you. It must disappeared into the Microsoft digital universe. I do hope to hear from you soon.

Thanks and regards 
Rupert

**Comment by jp-weber** [comment]
Hello,

Have you already tried uninstalling and reinstalling the app? According to some users, this behaviour occurs when using applications like Wise Disk Cleaner or CCleaner are used which do more harm than good.

Regarding the second behaviour, is the time and region correct? Some services have a very short time window in which the generated codes are valid. For a potential log entry I would need the log entries, these can be opened directly with the following command in the cmd/powershell: https://github.com/2fast-team/2fast/wiki/Open-(error)-log-via-cmd

**Comment by ZupflKnoe568del** [comment]
Hello Jan Philipp,thanks a lot for your super fast reply.I uninstalled all WISE programs I uninstalled 2fast and re-installed it.I synchronized to time.windows.comI rebooted Now it seems to be running stable. GREAT!As I already wrote: I subscribed to your program on a monthly basis BUT 2fast did not registered any subscription I paid via Microsoft account and PayPal money and I think the one-time fee but I really doubt that it ever reached you.Is it possible for you to check?I would like to avoid to see my money go down the drain of the digital nirvana.Thanks again for your help.Kind regards Rupert Am 02.11.24, 10:34 schrieb Jan Philipp Weber ***@***.***>:


Hello,
Have you already tried uninstalling and reinstalling the app? According to some users, this behaviour occurs when using applications like Wise Disk Cleaner or CCleaner are used which do more harm than good.
Regarding the second behaviour, is the time and region correct? Some services have a very short time window in which the generated codes are valid. For a potential log entry I would need the log entries, these can be opened directly with the following command in the cmd/powershell: https://github.com/2fast-team/2fast/wiki/Open-(error)-log-via-cmd

**Comment by jp-weber** [comment]
Great to hear, that the app problems are fixed 👍

The topic of subscriptions was somehow overlooked in the first post. Many thanks for your support! 

I have looked and found currently no monthly subscription in my incomming payments.  If it was a one-time payment, you should be able to see it in the settings under about the app/subscription and purchases.
![grafik](https://github.com/user-attachments/assets/7f117aca-0567-4b20-9523-ec3f9b053176)


The last monthly subscription is many month ago. I would suggest that you look at your subscription overview in your Microsoft account: https://account.microsoft.com/services

**Comment by ZupflKnoe568del** [comment]
Hi Jan Philipp,unfortunately, the joy didn't last:Not for first time I received this error message (in German): I'm translating into English;System.Exception: The file, which should have been replaced cannot be removed. (Exception from HRESULT:0x80070497)Can you please look into this matter?I do appreciate it.regards Rupert Am 03.11.24, 15:50 schrieb Jan Philipp Weber ***@***.***>:


Great to hear, that the app problems are fixed 👍
The topic of subscriptions was somehow overlooked in the first post. Many thanks for your support!
I have looked and found currently no monthly subscription in my incomming payments.  If it was a one-time payment, you should be able to see it in the settings under about the app/subscription and purchases.
grafik.png (view on web)
The last monthly subscription is many month ago. I would suggest that you look at your subscription overview in your Microsoft account: https://account.microsoft.com/services

**Reply by jp-weber** [reply]
The error occurs when changes cannot be written to the file. Is it perhaps in a cloud storage (OneDrive, Google) etc.?

**Comment by ZupflKnoe568del** [comment]
2024-11-03 11:41:24 - uptime -10675199.02:48:05.4775808
2024-11-03 12:07:55 - uptime -10675199.02:48:05.4775808
2024-11-04 18:07:13 - uptime -10675199.02:48:05.4775808
2024-11-04 18:07:13 - <unknown> - Padding is invalid and cannot be removed.
Log - System.Security.Cryptography.CryptographicException: Padding is invalid and cannot be removed.
   at Internal.Cryptography.UniversalCryptoDecryptor.DepadBlock(Byte[], Int32, Int32) + 0x27a
   at Internal.Cryptography.UniversalCryptoDecryptor.UncheckedTransformFinalBlock(Byte[], Int32, Int32) + 0xda
   at Internal.Cryptography.UniversalCryptoTransform.TransformFinalBlock(Byte[], Int32, Int32) + 0x33
   at System.Security.Cryptography.CryptoStream.FlushFinalBlock() + 0x2e
   at System.Security.Cryptography.CryptoStream.Dispose(Boolean) + 0x23
   at System.IO.Stream.Close() + 0x13
   at System.IO.StreamReader.Dispose(Boolean) + 0x3d
   at System.Diagnostics.Tracing.EventSource.Dispose() + 0x10
   at Newtonsoft.Json.Encryption.Encrypter.Decrypt(String) + 0x130
   at StringConverter.ReadJson(JsonReader, Type, Object, JsonSerializer) + 0x33
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter, JsonReader, Type, Object) + 0x11b
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty, JsonConverter, JsonContainerContract, JsonProperty, JsonReader, Object) + 0x117
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object, JsonReader, JsonObjectContract, JsonProperty, String) + 0x6cd
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader, Type, JsonContract, JsonProperty, JsonContainerContract, JsonProperty, Object) + 0x32a
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader, Type, JsonContract, JsonProperty, JsonContainerContract, JsonProperty, Object) + 0xbc
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList, JsonReader, JsonArrayContract, JsonProperty, String) + 0x3de
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader, Type, JsonContract, JsonProperty, Object, String) + 0x110
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader, Type, JsonContract, JsonProperty, JsonContainerContract, JsonProperty, Object) + 0xf5
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty, JsonConverter, JsonContainerContract, JsonProperty, JsonReader, Object) + 0x16d
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object, JsonReader, JsonObjectContract, JsonProperty, String) + 0x6cd
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader, Type, JsonContract, JsonProperty, JsonContainerContract, JsonProperty, Object) + 0x32a
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader, Type, JsonContract, JsonProperty, JsonContainerContract, JsonProperty, Object) + 0xbc
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader, Type, Boolean) + 0x239
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader, Type) + 0xf3
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader) + 0x3b
   at Project2FA.Core.Services.JSON.NewtonsoftJSONService.DeserializeDecrypt[T](Byte[], Byte[], String, Int32) + 0x128
   at Project2FA.ViewModels.FileActivationPageViewModel.<TestPassword>d__19.MoveNext() + 0x21f

2024-11-04 18:07:13 - uptime -10675199.02:48:05.4775808

**Reply by jp-weber** [reply]
The error with "Padding is invalid and cannot be removed." is related to a wrong password. At which point is the error happened? After login, editing an account etc.?

**Comment by ZupflKnoe568del** [comment]
Hello Jan Philipp,password problem? Sorry, but 2fast used to tell me "Wrong password" when I put the password in wrongly like thousands of other programs would do. But for a while it simply throws me out right like bolt lightning after I put in my password. Right or wrong password. No message whatsoever. It just crashes on me in a fraction of a second.I know my 2fast password very well because I have to put it in two or three til I can use it relatively stably.And then the error message pops up that I sent you. And the app.log etc.If 2fast crashes during a transaction then I doubt if it's wise to keep using it. In the last 15 minutes it crashed on me 4 times. No other program does that on my Windows machine.By the way, my 2fast one-time Microsoft/Payment went through....Kind regards Rupert Am 05.11.24, 18:44 schrieb Jan Philipp Weber ***@***.***>:


The error with "Padding is invalid and cannot be removed." is related to a wrong password. At which point is the error happened? After login, editing an account etc.?

**Reply by jp-weber** [reply]
Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.

There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)? 
See wiki and: 
![grafik](https://github.com/user-attachments/assets/59d34291-6dfe-47cf-b3f5-b152865e5a8d)


The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.

Many thanks for your support!

**Comment by ZupflKnoe568del** [comment]
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!

**Comment by ZupflKnoe568del** [comment]
Hello Jan Philipp,sorry, but it's gotten worse. Normally, I had to open your 2fast 5 times which was ok and it worked but NOW it simply shows the yellow key with NO login boxes, and simply exits. And I repeat this 20-30 times to no avail. Sometimes rebooting helped but now NOTHING helps. I'm disappointed and frustrated.If there's no solution within the next 7 days I'm going to uninstall 2fast Authenticator.Regards Rupert Am 08.11.24, 18:51 schrieb RL ***@***.***>:
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!

**Comment by ZupflKnoe568del** [comment, marked as answer]
Hi JAN, EVERYTHING SEEMS TO BE OKAY NOW!!I uninstalled 2fast, downloaded it again from the MS Store AND IT WORKS faster than ever before, even the LoginI guess it was a MS Update issue.Have a good dayRupert Am 15.02.26, 16:03 schrieb RL ***@***.***>:
Hello Jan Philipp,sorry, but it's gotten worse. Normally, I had to open your 2fast 5 times which was ok and it worked but NOW it simply shows the yellow key with NO login boxes, and simply exits. And I repeat this 20-30 times to no avail. Sometimes rebooting helped but now NOTHING helps. I'm disappointed and frustrated.If there's no solution within the next 7 days I'm going to uninstall 2fast Authenticator.Regards Rupert Am 08.11.24, 18:51 schrieb RL ***@***.***>:
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!

**Comment by ZupflKnoe568del** [comment]
Hi Jan,until yesterday every was fine with 2fast but now I receive from different logins: “2fa 6-digit code is wrong." Is there something with the time servertime.windows.com?I need your support!thanks a lotRupert Am 16.02.26, 17:13 schrieb RL ***@***.***>:
Hi JAN, EVERYTHING SEEMS TO BE OKAY NOW!!I uninstalled 2fast, downloaded it again from the MS Store AND IT WORKS faster than ever before, even the LoginI guess it was a MS Update issue.Have a good dayRupert Am 15.02.26, 16:03 schrieb RL ***@***.***>:
Hello Jan Philipp,sorry, but it's gotten worse. Normally, I had to open your 2fast 5 times which was ok and it worked but NOW it simply shows the yellow key with NO login boxes, and simply exits. And I repeat this 20-30 times to no avail. Sometimes rebooting helped but now NOTHING helps. I'm disappointed and frustrated.If there's no solution within the next 7 days I'm going to uninstall 2fast Authenticator.Regards Rupert Am 08.11.24, 18:51 schrieb RL ***@***.***>:
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!

**Comment by ZupflKnoe568del** [comment]
Hi Jan,sorry to bother you again but I need urgent access to three important accunts. I need your support.Thanks a lot and regards.Rupert Am 25.04.26, 11:19 schrieb RL ***@***.***>:
Hi Jan,until yesterday every was fine with 2fast but now I receive from different logins: “2fa 6-digit code is wrong." Is there something with the time servertime.windows.com?I need your support!thanks a lotRupert Am 16.02.26, 17:13 schrieb RL ***@***.***>:
Hi JAN, EVERYTHING SEEMS TO BE OKAY NOW!!I uninstalled 2fast, downloaded it again from the MS Store AND IT WORKS faster than ever before, even the LoginI guess it was a MS Update issue.Have a good dayRupert Am 15.02.26, 16:03 schrieb RL ***@***.***>:
Hello Jan Philipp,sorry, but it's gotten worse. Normally, I had to open your 2fast 5 times which was ok and it worked but NOW it simply shows the yellow key with NO login boxes, and simply exits. And I repeat this 20-30 times to no avail. Sometimes rebooting helped but now NOTHING helps. I'm disappointed and frustrated.If there's no solution within the next 7 days I'm going to uninstall 2fast Authenticator.Regards Rupert Am 08.11.24, 18:51 schrieb RL ***@***.***>:
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!

**Comment by ZupflKnoe568del** [comment]
Hi Jan,I just synced the Windows time. Now 2fast generates the codes correctly. 👍Have a good day.Regards Rupert Am 28.04.26, 07:58 schrieb RL ***@***.***>:
Hi Jan,sorry to bother you again but I need urgent access to three important accunts. I need your support.Thanks a lot and regards.Rupert Am 25.04.26, 11:19 schrieb RL ***@***.***>:
Hi Jan,until yesterday every was fine with 2fast but now I receive from different logins: “2fa 6-digit code is wrong." Is there something with the time servertime.windows.com?I need your support!thanks a lotRupert Am 16.02.26, 17:13 schrieb RL ***@***.***>:
Hi JAN, EVERYTHING SEEMS TO BE OKAY NOW!!I uninstalled 2fast, downloaded it again from the MS Store AND IT WORKS faster than ever before, even the LoginI guess it was a MS Update issue.Have a good dayRupert Am 15.02.26, 16:03 schrieb RL ***@***.***>:
Hello Jan Philipp,sorry, but it's gotten worse. Normally, I had to open your 2fast 5 times which was ok and it worked but NOW it simply shows the yellow key with NO login boxes, and simply exits. And I repeat this 20-30 times to no avail. Sometimes rebooting helped but now NOTHING helps. I'm disappointed and frustrated.If there's no solution within the next 7 days I'm going to uninstall 2fast Authenticator.Regards Rupert Am 08.11.24, 18:51 schrieb RL ***@***.***>:
Hello Jan Philipp,I entered the (Anmeldeinformationsverwaltung),punched in my Windows logon credentials,made my 2fast password visible, which was/is definitely the correct on, and started the 2fast.2fa file from the Explorer... and...It worked.However, for testing reasons, I deliberately put in a wrong password and 2fast kicked me out again with no comment at all.I then started it again with no problem and the 6-digit codes are in sync that's the main thing for me, Jan.Let's leave it at that.I do appreciate your superb, competent, and fast support.Have a great weekend.Kind regards Rupert Am 07.11.24, 20:30 schrieb Jan Philipp Weber ***@***.***>:


Hello Rupert, that sounds first strange with the password problem, but it is indeed the case, which is why Wise disc cleaner also played a role here. With the configuration of the data file in the first step of the app, the entered password is stored in a protected memory in Windows (see wiki https://github.com/2fast-team/2fast/wiki/View-password-for-encryption-on-local-device) and is retrieved from there by the authentication every time, the password is needed. This is build so, that the password entered at login is not in plain text during the runtime in the memory but is retrieved from the protected memory.
There are two things to test here, is the entry with the password available in the windows credential store (Anmeldeinformationsverwaltung)?
See wiki and:
grafik.png (view on web)
The second thing to test is how the file is started with a double click in Windows Explorer and whether this works without problems. A different method is used here.
Many thanks for your support!
